### PR TITLE
Add twitter card image support

### PR DESCRIPTION
**Pull request by VagishVela**
So far started working on it, just creating a PR to track status of #933.

- [x] Add support for local paths
- [x] Add support for non-local paths
- [x] Account for protocol-less urls
- [ ] Move regexp test to the loader
- [ ] Appropriate tests

### CODE REVIEW

**Comment by MoOx** [comment]
Nice start!

Could you add as well og support?

**Comment by VagishVela** [comment]
@MoOx Sure thing.

File: themes/phenomic-theme-base/src/layouts/Page/index.js [hunk 1]
```diff
@@ -30,22 +30,24 @@ const Page = (
 
   const metaTitle = head.metaTitle ? head.metaTitle : head.title
 
+  const externalUrlCheck = new RegExp('^https?://');
+  const socialImage = externalUrlCheck.test(head.hero) ?
+  head.hero : joinUri(process.env.PHENOMIC_USER_URL, head.hero)
+
```
Comment (thangngoc89) [inline themes/phenomic-theme-base/src/layouts/Page/index.js]: I think we  should move this into the loader part, processing this on every render isn't good for performance wise

Comment (thangngoc89) [inline themes/phenomic-theme-base/src/layouts/Page/index.js]: Also, i usually use protocol-less url (//example.com/img.jpg)

**Comment by thangngoc89** [comment]
Also, the process.env.PHENOMIC_USER_URL is always what you defined in `package.json` homepage url. Please handle the development mode.

**Comment by MoOx** [comment]
We should probably handle that. The theme should not have code for development imo.

**Comment by thangngoc89** [comment]
So PHENOMIC_URL. I guess? But the port can't be known before starting the dev server

**Comment by MoOx** [comment]
We should not move anything to the loader as it's going to disappear.

**Comment by VagishVela** [comment]
@MoOx @thangngoc89 So where should I handle https://github.com/VagishVela/phenomic/blob/261c9cc08e8c4d3b6f38f20e296171ce709e5b0b/themes/phenomic-theme-base/src/layouts/Page/index.js#L33-L35 ?

File: themes/phenomic-theme-base/src/layouts/Page/index.js [hunk 2]
```diff
@@ -30,18 +30,24 @@ const Page = (
 
   const metaTitle = head.metaTitle ? head.metaTitle : head.title
 
+  const externalUrlCheck = new RegExp('^(https?:)?\/\/');
+  const socialImage = externalUrlCheck.test(head.hero) ?
+  head.hero : joinUri(process.env.PHENOMIC_USER_URL, head.hero)
```
Comment (MoOx) [inline themes/phenomic-theme-base/src/layouts/Page/index.js]: Could you simplify to

```js
const socialImage = head.hero.match("://") ? head.hero
  : joinUri(process.env.PHENOMIC_USER_URL, head.hero)
```

**Comment by MoOx** [comment]
Thanks for this contribution!

**Pull request merged**